tlspuffin/src/openssl/deterministic.rs

```diff
@@ -1,35 +1,53 @@
 use std::os::raw::c_int;
 
-use log::warn;
+use log::{trace, warn};
 
 #[cfg(feature = "deterministic")]
 extern "C" {
     fn make_openssl_deterministic();
     fn RAND_seed(buf: *mut u8, num: c_int);
+    pub static mut tlspuffin_seed: u32;
 }
 
+pub fn get_seed() -> u32 {
+    unsafe { return tlspuffin_seed }
+}
 #[cfg(feature = "deterministic")]
-pub fn set_openssl_deterministic() {
-    warn!("OpenSSL is no longer random!");
+pub fn determinism_set_reseed_openssl() {
+    trace!("Making OpenSSL fully deterministic: reset rand and reseed to a constant...");
     unsafe {
         make_openssl_deterministic();
+    }
+    determinism_reseed_openssl();
+}
+
+#[cfg(feature = "deterministic")]
+pub fn determinism_reseed_openssl() {
+    trace!(" - Reseed RAND for OpenSSL");
+    unsafe {
         let mut seed: [u8; 4] = 42u32.to_le().to_ne_bytes();
         let buf = seed.as_mut_ptr();
         RAND_seed(buf, 4);
+        tlspuffin_seed = 42 as u32;
     }
 }
 
 #[cfg(test)]
 mod tests {
     use openssl::rand::rand_bytes;
 
+    use crate::openssl::deterministic::{determinism_set_reseed_openssl, get_seed};
+
     #[test]
     #[cfg(feature = "openssl111-binding")]
-    fn test_openssl_no_randomness() {
-        use crate::openssl::deterministic::set_openssl_deterministic;
-        set_openssl_deterministic();
+    fn test_openssl_no_randomness_simple() {
+        determinism_set_reseed_openssl();
+        assert_eq!(get_seed(), 42);
         let mut buf1 = [0; 2];
         rand_bytes(&mut buf1).unwrap();
-        assert_eq!(buf1, [183, 96]);
+        assert_eq!(buf1, [177, 180]);
+        assert_ne!(get_seed(), 42);
+        determinism_set_reseed_openssl();
+        assert_eq!(get_seed(), 42);
```

Comment:
Why did that change?

Reply:
Had to do this accoridnly to the the changes made in [crates/openssl-src-111/src/deterministic_rand.c](https://github.com/tlspuffin/tlspuffin/pull/290/files/8865f8b83db41dc5cf9ac7c220f2711d0b01e8ef#diff-ff8a9ea809dc3abf285d6dfdeaf19161e1847aaf1c6102ac2c345639ba639c8f). As we revert the later, we will revert this as well.